salt/kubelet/stop.sls

```diff
@@ -4,6 +4,8 @@
 include:
   - kubectl-config
 
+{% set should_uncordon = salt['cmd.run']("kubectl --kubeconfig=" + pillar['paths']['kubeconfig'] + " get nodes " + grains['caasp_fqdn'] + " -o=jsonpath='{.spec.unschedulable}' 2>/dev/null") != "true" %}
+
 # If this fails we should ignore it and proceed anyway as Kubernetes will recover
 drain-kubelet:
```

Comment:
Shouldn't we do this only if the node is currently _scheduable_?

Reply:
Yes, that's what we are doing, `.spec.unschedulable != true`.

Reply:
If you are talking about the drain specifically, we need to drain. A `unschedulable` node will not accept new workloads, but will happily keep workloads that were running before it was set as `unschedulable`. The drain is necessary always.

Reply:
Then how can we detect if the node has already been drained?

Reply:
There's no need to check if a node has been drained. You want to drain the node, drain will mark the node as unschedulable and evict all running pods. If it was drained before there will be no difference, this operation will succeed.

Reply:
By the way, will this mark master nodes as _scheduable_ after the reboot? Because the are _unscheduable_, so this will `drain` them and mark as `should_uncordon=True`, and then they will be `undordon`ed after reboot, so I don't know if hey will be marked `scheduable`...

Reply:
`kubelet.stop` will only be executed on the worker nodes. Also, we never ever uncordon master nodes (https://github.com/kubic-project/salt/blob/master/salt/kubelet/init.sls#L84)

Reply:
> kubelet.stop will only be executed on the worker nodes. Also, we never ever uncordon master nodes (https://github.com/kubic-project/salt/blob/master/salt/kubelet/init.sls#L84)

Ok, makes sense.

Anyway, this will not skip the `cordon` when the node has already be `cordon`ed by users, but maybe not a big deal...